src/crimson/os/seastore/extent_placement_manager.h

```diff
@@ -0,0 +1,400 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:nil -*-
+// vim: ts=8 sw=2 smarttab expandtab
+
+#pragma once
+
+#include "seastar/core/gate.hh"
+
+#include "crimson/common/condition_variable.h"
+#include "crimson/common/log.h"
+#include "crimson/os/seastore/cache.h"
+#include "crimson/os/seastore/cached_extent.h"
+#include "crimson/os/seastore/lba_manager.h"
+
+namespace crimson::os::seastore {
+
+/**
+ * ool_record_t
+ *
+ * Encapsulates logic for building and encoding an ool record destined for
+ * an ool segment.
+ *
+ * Uses a metadata header to enable scanning the ool segment for gc purposes.
+ * Introducing a seperate physical->logical mapping would enable removing the
+ * metadata block overhead.
+ */
+class ool_record_t {
+  class OolExtent {
+  public:
+    OolExtent(LogicalCachedExtentRef& lextent)
+      : lextent(lextent) {}
+    void set_ool_paddr(paddr_t addr) {
+      ool_offset = addr;
+    }
+    paddr_t get_ool_paddr() const {
+      return ool_offset;
+    }
+    void persist_paddr() {
+      lextent->set_paddr(ool_offset);
+      ool_offset = P_ADDR_NULL;
+    }
+    bufferptr& get_bptr() {
+      return lextent->get_bptr();
+    }
+    LogicalCachedExtentRef& get_lextent() {
+      return lextent;
+    }
+  private:
+    paddr_t ool_offset;
+    LogicalCachedExtentRef lextent;
+  };
+
+public:
+  ool_record_t(size_t block_size) : block_size(block_size) {}
+  record_size_t get_encoded_record_length() {
+    return crimson::os::seastore::get_encoded_record_length(record, block_size);
+  }
+  size_t get_wouldbe_encoded_record_length(LogicalCachedExtentRef& extent) {
+    auto raw_mdlength = get_encoded_record_raw_mdlength(record, block_size);
+    auto wouldbe_mdlength = p2roundup(
+      raw_mdlength + ceph::encoded_sizeof_bounded<extent_info_t>(),
+      block_size);
+    return wouldbe_mdlength + extent_buf_len + extent->get_bptr().length();
+  }
+  ceph::bufferlist encode(segment_id_t segment, segment_nonce_t nonce) {
+    assert(extents.size() == record.extents.size());
+    auto rsize = get_encoded_record_length();
+    segment_off_t extent_offset = base + rsize.mdlength;
+    for (auto& extent : extents) {
+      extent.set_ool_paddr(
+        {segment, extent_offset});
+      extent_offset += extent.get_bptr().length();
+    }
+    return encode_record(rsize, std::move(record), block_size, nonce);
```

Comment:
I only find 
ceph::bufferlist encode_record(
  record_size_t rsize,
  record_t &&record,
  size_t block_size,
  segment_off_t committed_to,
  segment_nonce_t current_segment_nonce = 0);

so here  return encode_record(rsize, std::move(record), block_size, nonce);  the fourth parameter nonce match the committed_to ?

Reply:
This should be a bug, it hasn't caused any problem probably because ool record's committed_to is not used by far. Thanks:-)